src/hm-search.js

```diff
@@ -276,25 +292,35 @@ function compileIndexTypeVariables(item) {
   return item;
 }
 
-function index(sigs) {
+function buildDb(sigs) {
   return (sigs
     .map(indexParse)
     .map(when(isMethod, methodToFunction))
     .map(compileIndexTypeVariables)
   );
 }
 
-function search(db, input) {
-  var parsed = searchParse(input);
+function init(data, opts) {
+  if (opts == null) {
+    opts = {
+      fuzzy: true,
+    };
+  }
+  var db = buildDb(data);
+  return {
+    search: function search(input) {
+      var parsed = searchParse(input);
 
-  db = nameSearch(db, parsed.name);
-  db = typeSearch(db, parsed.type);
+      var result = db;
+      result = nameSearch(result, parsed.name, opts);
+      result = typeSearch(result, parsed.type);
 
-  return db;
+      return result;
```

Comment:
I find this reassignment slightly upsetting. :stuck_out_tongue_winking_eye:

I would prefer to nest the expressions:

```javascript
return typeSearch(nameSearch(db, parsed.name, opts), parsed.type);
```